ResearchKit/Common/ORKReviewStep.md

```diff
@@ -0,0 +1,81 @@
+<h2>Introduce</h2>
+- `ORKReviewStep`
+- `ORKReviewStepViewController`
+- `ORKReviewStepViewControllerDelegate`
+- `ORKReviewStepReviewDirection`
+
+<h2>Change</h2>
+- `ORKTaskViewController`
+
+<h2>ORKReviewStep</h2>
+<h3>properties</h3>
+    NSArray *steps
+    ORKTaskResultSource *resultSource
+    ORKReviewStepReviewDirection *reviewDirection
+<h3>use cases</h3>
+- __outside of a survey = standalone__
+  - steps array is provided upon initialization
+  - review step is at the beginning
+  - step results may not be changed
+  - review direction defaults to forward
+		
+- __within a survey__
+  - steps array is nil
+  - review step is at the end or in the middle
+  - step results can be changed
+  - review direction defaults to reverse			
+
+<h2>ORKReviewStepViewControllerDelegate</h2>
+<h3>methods</h3>
+    reviewStepViewController: canReviewStep: 
+        called to determine if a step can be reviewed
+    
+    reviewStepViewController: canChangeStep: 
+        called to determine if an answer can be changed during review
+    
+    reviewStepViewController: reviewStep: 
+        called when a step is selected for review
+
+<h2>ORKReviewStepViewController</h2>
+<h3>properties</h3>
+    NSArray *steps
+    ORKTaskResultSource *resultSource
+    ORKReviewStepViewControllerDelegate *reviewDelegate
+    ORKReviewStepReviewDirection reviewDirection
+<h3>methods</h3>
+    discoverSteps: withResult:
+            repeatedly call stepAfterStep: (reviewDirection forward) or stepBeforeStep: (reviewDirection reverse) until upcoming step is nil or another review step is reached
+            for every step in steps array
+                call reviewStepViewController: shouldIncludeStep:
+            reload tableView data			
+
+    tableView: didSelectRowAtIndexPath:
+        if reviewStepViewController: canChangeStep: returns true, call reviewStepViewController: reviewStep:
```

Comment:
Okay, that will require some changes to `ORKStepViewController` as well. I'd suggest a `canChangeStepResult` property. If set to `YES` the behavior is just as it is now. If set to `NO` we need to disable any UI elements that may change the result. Furthermore, `ORKTaskViewController` should not add the result of an immutable step to its' `managedResults` array.